app/src/main/java/com/asfoundation/wallet/referrals/ReferralsFragment.kt

```diff
@@ -0,0 +1,145 @@
+package com.asfoundation.wallet.referrals
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.View.GONE
+import android.view.View.VISIBLE
+import android.view.ViewGroup
+import com.asf.wallet.R
+import dagger.android.support.DaggerFragment
+import kotlinx.android.synthetic.main.invited_friends_animation_list.*
+import kotlinx.android.synthetic.main.referrals_layout.*
+import java.math.BigDecimal
+import java.math.RoundingMode
+import kotlin.math.roundToInt
+
+class ReferralsFragment : DaggerFragment(), ReferralsView {
+
+  private lateinit var presenter: ReferralsPresenter
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    presenter = ReferralsPresenter(this)
+  }
+
+  override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+    super.onViewCreated(view, savedInstanceState)
+    presenter.present()
+  }
+
+  override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
+                            savedInstanceState: Bundle?): View? {
+    return inflater.inflate(R.layout.referrals_layout, container, false)
+  }
+
+  override fun setupLayout() {
+    val totalAvailable = completedInvites + available
+    friends_invited.text = String.format("%d/%d", completedInvites, totalAvailable)
+    friends_invited.visibility = VISIBLE
+    number_friends_invited.text = String.format("%d/%d", completedInvites, totalAvailable)
+    total_earned.text = currency.plus(receivedAmount)
+    total_earned.visibility = VISIBLE
+    val individualEarn = currency + amount.setScale(2, RoundingMode.FLOOR).toString()
+    val totalEarn = currency + maxAmount.setScale(2, RoundingMode.FLOOR).toString()
+    referral_explanation.text =
+        getString(R.string.referral_dropup_menu_requirements_body, individualEarn, totalEarn)
+    invitations_progress_bar.progress =
+        ((100 / (completedInvites.toDouble() + available.toDouble())) * completedInvites).roundToInt()
+    setFriendsAnimations(completedInvites, completedInvites + available)
+  }
+
+  private fun setFriendsAnimations(invited: Int, totalInvitations: Int) {
+    val friendsAnimation =
+        arrayOf(friend_animation_1, friend_animation_2, friend_animation_3, friend_animation_4,
+            friend_animation_5)
+    if (invited != 0) {
+      for (i in friendsAnimation.indices) {
+        if (i < invited) {
+          friendsAnimation[i].setAnimation(R.raw.invited_user_animation)
+          friendsAnimation[i].playAnimation()
+        }
+        if (i >= totalInvitations) friendsAnimation[i].visibility = GONE
+      }
+    } else if (totalInvitations < friendsAnimation.size) {
```

Comment:
After looking at this again, it doesn’t need any changes because of the “if” before the else if (if (i >= totalInvitations)) which will remove the icons if the animation index is bigger than the number of invitations.
I’ll still improve the method as it was proven that it can be hard to understand